test/contrasts.jl

```diff
@@ -0,0 +1,92 @@
+module TestContrasts
+
+using Base.Test
+using DataFrames
+
+
+d = DataFrame(x = @pdata( [:a, :b, :c, :a, :a, :b] ))
+
+mf = ModelFrame(Formula(Nothing(), :x), d)
+
+@test ModelMatrix(mf).m == [1  0  0
+                            1  1  0
+                            1  0  1
+                            1  0  0
+                            1  0  0
+                            1  1  0]
+@test coefnames(mf) == ["(Intercept)"; "x: b"; "x: c"]
+
+contrast!(mf, x = SumContrasts)
+@test ModelMatrix(mf).m == [1 -1 -1
+                            1  1  0
+                            1  0  1
+                            1 -1 -1
+                            1 -1 -1
+                            1  1  0]
+@test coefnames(mf) == ["(Intercept)"; "x: b"; "x: c"]
```

Comment:
For sum contrasts, I think it is more common to omit the *last* level by default. Do you think it would be possible without too much work?